src/request/index.js

```diff
@@ -0,0 +1,60 @@
+var http = require('http')
```

Comment:
All this stuff will make it's way into Webpack build since this file is directly referenced in `client-oauth2.js` ... which will bloat the build.

The correct way is to make 2 separate entry points in `package.json`: `main` and `browser`. In those entry points you need to configure which implementation of `request` to use. Also, by the way, you can allow to provide `request` implementation via DI just like Promise.